criu/files-reg.c

```diff
@@ -2115,19 +2140,36 @@ int open_path(struct file_desc *d, int (*open_cb)(int mntns_root, struct reg_fil
 
 	if (rfi->remap) {
 		if (!rfi->remap->is_dir) {
-			unlinkat(mntns_root, rfi->path, 0);
-			rm_parent_dirs(mntns_root, rfi->path, level);
+			struct mount_info *mi = lookup_mnt_id(rfi->rfe->mnt_id);
+
+			if (mi && try_remount_writable(mi, true))
+				goto err;
+
+			pr_debug("%d: Unlink: %s\n", rfi->rfe->mnt_id, rfi->path);
+			if (unlinkat(mntns_root, rfi->path, 0)) {
+				pr_perror("Failed to unlink the remap file");
```

Comment:
`pr_perror("Failed to unlink the remap file %d:%s", mntns_root, rfi->path);`

Reply:
- mntns_root is a ROOT_FD_OFF service fd number, I believe printing it has not much value
- the pr_debug above already prints mnt_id+path so we can skip it in pr_perror

Reply:
Log messages can get interleaved due to parallelism. It could br useful to have a more precise error message.

Reply:
It's very common practice in criu that not only the error itself should be considered when debugging but also previous log messages. I get used to this practice probably too much =) So I would've just grep logs by pid to neutralize interleaved messages. But OK I can just make error message printing similar thing to pr_debug.

Reply:
Fair, I don't feel strongly about this one, your call

Reply:
So let's revert it to my variant then.